llvm/include/llvm/ProfileData/PGOCtxProfReader.h

```diff
@@ -115,6 +118,9 @@ class PGOCtxProfContext final : public internal::IndexNode {
   const SmallVectorImpl<uint64_t> &counters() const { return Counters; }
   SmallVectorImpl<uint64_t> &counters() { return Counters; }
 
+  bool isRoot() const { return RootEntryCount.has_value(); }
+  uint64_t getTotalRootEntryCount() const { return *RootEntryCount; }
```

Comment:
Dereferencing std::optional using operator* when it doesn't have a value is undefined behaviour. Prefer `.value()` which throws an exception (or fails with -fno-exception) or maybe add a `has_value()` check for value and return a default?


https://en.cppreference.com/w/cpp/utility/optional/operator*

Reply:
thanks, done.